Fix digital_ocean module_util api_token bug

BondAnthony opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Update the self.module.params to a get('key') to prevent an exception when the param cannot be found.
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
module_util/digital_ocean.py
##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible 2.4.0 (devel ae6d27fa29) last updated 2017/09/01 08:43:30 (GMT -400)
  config file = /Users/devMac/ansible.cfg
  configured module search path = [u'/Users/devMac/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/devMac/Development/github/ansible/lib/ansible
  executable location = /Users/devMac/Development/github/ansible/bin/ansible
  python version = 2.7.10 (default, Oct 23 2015, 19:19:21) [GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)]
```
This PR addresses issue: #28923

BondAnthony committed:
Fix digital_ocean module_util api_token bug

BondAnthony commented:
@gundalow if you have a couple minutes could you please review this. I've tested this change but I'm not entirely sure if this would be the correct way to address my problem.

When the ```get_do_oauth_token``` is called within module_utils/digital_ocean.py it's failing to find ```api_token``` parameter because an exception is thrown on ```self.module.params.get('oauth_token')``` since it's undefined. The module should be able to find either parameter or three different environment variables.

....Yes I need to clean up the variables and get down to a single variable.

s-hertel commented on lib/ansible/module_utils/digital_ocean.py:
```diff
@@ -93,8 +93,8 @@ def delete(self, path, data=None):
-            self.oauth_token = self.module.params['oauth_token'] or \
-                self.module.params['api_token'] or \
+            self.oauth_token = self.module.params.get('oauth_token') or \
+                self.module.params.get('api_token') or \
```
Should the environment variables also be looked for with get()?

s-hertel replied:
That's how AWS credentials are handled too. https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ec2.py#L187

Besides that, this looks good to me.

BondAnthony replied:
Thank you I was looking for an example! I figured those would need to be included also but wasn't sure. New commit has been added with a change to the environment variables.

BondAnthony committed:
Included environment variables also

s-hertel commented:
@BondAnthony I don't think `except KeyError as e:` will ever happen now. If self.oauth_token is None at the end of this, should self.module.fail_json be called? Not sure if a lack of credentials is handled elsewhere for digital ocean.

BondAnthony commented:
@s-hertel you're right the module should probably call `fail_json` since we have exhausted all possible solutions for authentication to the API.

BondAnthony committed:
Removed try/catch and added a check on self.oauth_token

Modules using the DigitalOceanHelper would expect the module to handle any api key resolution.

BondAnthony commented:
@s-hertel take a peek at the newest commit and let me know what you think. Thanks!

Akasurde reviewed (approved):
shipit

BondAnthony commented:
+label digital_ocean

BondAnthony commented:
Hey @abadger would it be possible to get this merged into 2.4, of course when you have a few minutes?

abadger commented:
Merged to devel for inclusion in 2.4.0.  Thanks @BondAnthony for the fix and @s-hertel and @Akasurde for the reviews!

BondAnthony commented:
Thank you for the merge!

Thanks
Anthony

On Sep 6, 2017, at 10:15 AM, Toshio Kuratomi <notifications@github.com> wrote:

Merged to devel for inclusion in 2.4.0. Thanks @BondAnthony for the fix and @s-hertel and @Akasurde for the reviews!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.
